client/src/elements/schema-editor/schema-editor-geojson-point.js

```diff
@@ -106,23 +101,46 @@ export class SchemaEditorGeojsonPoint {
       scrollWheelZoom: 'center'
     }).setView([47.36521171867246, 8.547435700893404], 13);
 
-    // we need a mapzen key for the geocoder, this has to be defined in ENV
-    // if it's not here, we do not add a geocoder
-    const mapzenApiKey = await qEnv.mapzenApiKey;
-    if (mapzenApiKey === undefined) {
-      log.info('no mapzenApiKey defined, will not load geocoder for geojson-point editor');
+    if (!schemaEditorConfig.geojson.opencagedata.apiKey) {
+      log.info('no opencageApiKey given, will not load geocoder for geojson-point editor');
     } else {
-      this.geocoder = window.Leaflet.control.geocoder(mapzenApiKey, {
-        attribution: null,
-        fullWidth: true,
-        markers: false,
-        focus: false
+      this.geocoder = new window.Leaflet.Control.Search({
+        sourceData: async (text, next) => {
+          const data = await geocode(text, schemaEditorConfig.geojson.opencagedata.apiKey, schemaEditorConfig.geojson.opencagedata.language);
+          next(data);
+        },
+        filterData: (value, records) => {
+          return records; // do not filter the results
+        },
+        formatData: (data) => {
+          return data.features
+            .filter(result => {
+              return result.geometry.type === 'Point';
+            })
+            .map(result => {
+              return {
+                label: result.properties.formatted,
+                location: [result.geometry.coordinates[1], result.geometry.coordinates[0]]
+              };
+            })
+            .reduce((allResults, result) => {
+              allResults[result.label] = result.location;
+              return allResults;
+            }, {});
+        },
+        minLength: 3,
+        autoCollapse: true,
+        marker: false,
+        textPlaceholder: ''
       });
 
-      this.geocoder.on('select', e => {
-        this.geocoder.collapse();
-        this.data.geometry = e.feature.geometry;
-        this.data.properties.label = e.feature.properties.name;
+      this.geocoder.on('search:locationfound', (selectedSearchResult) => {
+        console.log(selectedSearchResult);
+        this.data.geometry = {
+          type: 'Point',
+          coordinates: [selectedSearchResult.latlng[1], selectedSearchResult.latlng[0]]
+        };
+        this.data.properties.label = selectedSearchResult.text;
```

Comment:
something seems to go wrong here, either selectedSearchResult.text is not working correctly or the label is not displayed correctly within the form

Reply:
It depends on the result as far as I have seen. unfortunately not all results have the `text` property. do you have an example? what you searched/selected?